io/src/main/scala/sbt/nio/Glob.scala

```diff
@@ -209,7 +209,8 @@ object Glob extends LowPriorityGlobOps {
     val sorted = globs.toSeq.sorted
     val needListDirectory: Path => Boolean = (path: Path) => {
       val filters = sorted.map(g => Glob(g.base, (0, g.range._2), AllPass).filter)
-      filters.exists(_.accept(path))
+      val a = path.resolve("a")
```

Comment:
What does `"a"` mean?

Reply:
I should add a comment. Say I have a Glob: `/foo/bar/*/*/*.txt` (which would have the representation `Glob(Paths.get("/foo.bar", (3, 3), "*.txt")` and I want to check if the path `/foo/bar/baz/` needs to be listed. To do this, I simplify the glob to `Glob(paths.get("/foo/bar/", (0, 3), AllPass). I then create the path `/foo/bar/baz/a`. Because that path has depth 2, it is included in the range which means that it is possible that there are children of `/foo/bar/baz` that might be accepted by `/foo/bar/*/*/*.txt`. If, on the other hand, the path was `/foo/bar/baz/buzz/banana`, the depth `/foo/bar/baz/buzz/banana/a` is 4 and it would be rejected by the simplified glob so we won't list that path.

Reply:
Writing that comment, I realized that I could rewrite this:
`sorted.exists(g => path.startsWith(g.base) && base.relativize(path).getNameCount < g.range._2).